src/doc/rustdoc/src/linking-to-items-by-name.md

```diff
@@ -24,11 +24,29 @@ pub struct Foo4;
 pub struct Bar;
 ```
 
+Unlike normal markdown, `[bar][Bar]` syntax is also supported without needing a
+`[Bar]: ...` reference link, and links are case-sensitive.
+
 Backticks around the link will be stripped, so ``[`Option`]`` will correctly
 link to `Option`.
 
-You can refer to anything in scope, and use paths, including `Self`, `self`,
-`super`, and `crate`. You may also use `foo()` and `foo!()` to refer to methods/functions and macros, respectively.
+## Valid links
+
+You can refer to anything in scope, and use paths, including `Self`, `self`, `super`, and
+`crate`. Associated items (functions, types, and constants) are supported, but [not for blanket
+trait implementations][#79682]. Rustdoc also supports linking to the following primitives, which
+have no path and cannot be imported:
+
+- [`slice`](https://doc.rust-lang.org/std/primitive.slice.html)
+- [`array`](https://doc.rust-lang.org/std/primitive.array.html)
+- [`tuple`](https://doc.rust-lang.org/std/primitive.tuple.html)
+- [`unit`](https://doc.rust-lang.org/std/primitive.unit.html)
+- [`fn`](https://doc.rust-lang.org/std/primitive.fn.html)
+- [`pointer`](https://doc.rust-lang.org/std/primitive.pointer.html), `*`, `*const`, or `*mut`
+- [`reference`](https://doc.rust-lang.org/std/primitive.reference.html), `&`, or `&mut`
+- [`never`](https://doc.rust-lang.org/std/primitive.never.html) or `!`
```

Comment:
I think we're supposed to use relative links in docs so that they refer to the same release channel. But these types don't change much, so probably not a big deal.

Reply:
Done, thanks.